### PR TITLE
DAT 3923 add custom pop over with delete item functionality to infobox builder preview

### PR DESCRIPTION
**Pull request by RafLeszczynski**
## Links

* https://wikia-inc.atlassian.net/browse/DAT-3923

## Description

add custom pop over with delete item functionality to infobox builder preview

## Reviewers

@dianafa @SebastianMarzjan @bognix @rogatty

### CODE REVIEW

**Comment by RafLeszczynski** [comment]
would also like to add two more integration test to `infobox-builder-edit-options-pop-over` component

File: front/main/app/styles/component/_infobox-builder.scss [hunk 1]
```diff
@@ -54,6 +55,16 @@ $spinner-text-color: rgba(0, 0, 0, .5);
 	@media (min-width: lower-bound($large-range)) {
 		padding-top: $wikia-ui-components-sub-head-height-large;
 	}
+
+	.infobox-builder-sidebar-header-icon-delete {
+		.icon-button-icon {
```
Comment (idradm) [inline front/main/app/styles/component/_infobox-builder.scss]: Are you sure about this name?

File: front/main/app/templates/components/infobox-builder.hbs [hunk 2]
```diff
@@ -38,6 +39,12 @@
 				message=(i18n 'main.reorder-tooltip' ns='infobox-builder')
 			}}
 		{{/if}}
+		{{infobox-builder-edit-options-pop-over
+			visible=isEditPopOverVisible
+			offsetX=15
+			onDeleteItem=(action this.attrs.removeItem)
```
Comment (hakubo) [inline front/main/app/templates/components/infobox-builder.hbs]: I think attrs.removeItem is enough
this is a default context in a template

File: front/main/app/templates/components/infobox-builder-edit-options-pop-over.hbs [hunk 3]
```diff
@@ -0,0 +1,13 @@
...
+		<div class="pop-over-container">
+			{{wikia-ui-components/icon-button
+				class='infobox-builder-sidebar-header-icon-delete'
+				click=(action 'removeItem')
+				icon='delete-large'
+				iconSize=24
+			}}
+		</div>
+	</div>
+{{/if}}
```
Comment (idradm) [inline front/main/app/templates/components/infobox-builder-edit-options-pop-over.hbs]: Please add new lines at the file ends

File: front/main/app/components/infobox-builder-edit-options-pop-over.js [hunk 4]
```diff
@@ -0,0 +1,102 @@
...
+		 * Sets data required for calculation pop over position based on DOM position
+		 * and dimensions on target element and pop over itself.
+		 * Need to do it on the DOM level after render to have access to rendered pop over DOM element dimensions.
+		 * @returns {void}
+		 */
+		setDataForCalculationComponentPosition() {
+			const $popOver = this.$(`.${this.get('popOverElementClassName')}`),
+				targetItem = this.get('targetItem'),
+				$targetItem = Ember.$(`#${targetItem.infoboxBuilderData.id}`),
+				position = $targetItem.position();
```
Comment (hakubo) [inline front/main/app/components/infobox-builder-edit-options-pop-over.js]: {left, top} = $targetItem.position();

File: front/main/app/components/infobox-builder-edit-options-pop-over.js [hunk 5]
```diff
@@ -0,0 +1,102 @@
...
+		popOverWidth: null,
+		popOverElementClassName: 'pop-over-compass',
+		didAttrsChange: false,
+		posX: Ember.computed('targetPosX', 'popOverWidth', 'offsetX', function () {
+			return this.get('targetPosX') - this.get('popOverWidth') - this.get('offsetX');
+		}),
+		posY: Ember.computed('targetPosY', 'targetHeight', 'popOverHeight', function () {
+			return this.get('targetPosY') + this.get('targetHeight') / 2 - this.get('popOverHeight') / 2;
+		}),
+		shouldUpdatePosition: Ember.computed.and('visible', 'didAttrsChange'),
```
Comment (hakubo) [inline front/main/app/components/infobox-builder-edit-options-pop-over.js]: this looks like a hack and it's hard to follow what is happening here.
why can't we update position on every didRender?

File: front/main/app/components/infobox-builder-edit-options-pop-over.js [hunk 6]
```diff
@@ -0,0 +1,102 @@
...
+		classNames: ['infobox-builder-edit-options-pop-over', 'pop-over', 'orient-left', 'pointer-center'],
+		targetPosX: null,
+		targetPosY: null,
+		targetHeight: null,
+		popOverHeight: null,
+		popOverWidth: null,
+		popOverElementClassName: 'pop-over-compass',
+		didAttrsChange: false,
+		posX: Ember.computed('targetPosX', 'popOverWidth', 'offsetX', function () {
+			return this.get('targetPosX') - this.get('popOverWidth') - this.get('offsetX');
```
Comment (idradm) [inline front/main/app/components/infobox-builder-edit-options-pop-over.js]: Could you init offsetX with 0 as default (+ tests)

File: front/main/app/templates/components/infobox-builder.hbs [hunk 7]
```diff
@@ -27,9 +29,16 @@
...
-				}}
-			{{/each}}
-		{{/sortable-group}}
+					}}
+				{{/each}}
+			{{/sortable-group}}
+			{{infobox-builder-edit-options-pop-over
+				visible=isEditPopOverVisible
+				offsetX=15
+				onDeleteItem=(action this.attrs.removeItem)
```
Comment (hakubo) [inline front/main/app/templates/components/infobox-builder.hbs]: I think attrs.removeItem is enough
this is a default context in a template

File: front/main/app/components/infobox-builder-edit-options-pop-over.js (hunk 4 above)
Comment (RafLeszczynski) [inline front/main/app/components/infobox-builder-edit-options-pop-over.js]: good catch :-)

File: front/main/app/templates/components/infobox-builder.hbs (hunk 7 above)
Comment (RafLeszczynski) [inline front/main/app/templates/components/infobox-builder.hbs]: if use use `attrs.removeItem` integration tests fail

so either `this.attrs.removeItem` or just `removeItem`

File: front/main/app/symbols/discussions/back-arrow.svg [hunk 8]
```diff
@@ -1,3 +0,0 @@
-<svg width="16" height="16" viewBox="0 0 16 16" xmlns="http://www.w3.org/2000/svg">
```
Comment (RafLeszczynski) [inline front/main/app/symbols/discussions/back-arrow.svg]: removal of duplicated arrow icon approved by Social Team. Talked with @kvas-damian about it.

Comment (kvas-damian) [inline front/main/app/symbols/discussions/back-arrow.svg]: :+1:
